ta/pkcs11/src/processing_asymm.c

```diff
@@ -308,10 +333,75 @@ static enum pkcs11_rc load_tee_key(struct pkcs11_session *session,
 }
 
 static enum pkcs11_rc
-init_tee_operation(struct pkcs11_session *session __unused,
-		   struct pkcs11_attribute_head *proc_params __unused)
+init_tee_operation(struct pkcs11_session *session,
+		   struct pkcs11_attribute_head *proc_params,
+		   struct pkcs11_object *obj)
 {
-	return PKCS11_CKR_OK;
+	enum pkcs11_rc rc = PKCS11_CKR_OK;
+	struct active_processing *proc = session->processing;
+	struct rsa_pss_processing_ctx *rsa_pss_ctx = NULL;
+	size_t modulus_size = 0;
+	size_t hash_size = 0;
+	uint32_t k = 0;
+
+	switch (proc_params->id) {
+	case PKCS11_CKM_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA1_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA224_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA256_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA384_RSA_PKCS_PSS:
+	case PKCS11_CKM_SHA512_RSA_PKCS_PSS:
+		rc = pkcs2tee_proc_params_rsa_pss(proc, proc_params);
```

Comment:
This is RSA specific (rsa/pss specific). Could you move this into a helper function (i.e. `pkcs2tee_proc_params_rsa_pss()`) in _processing_rsa.c_?

Reply:
I was thinking that `pkcs2tee_proc_params_rsa_pss()` would be more like translator and allocator for context.

But can move them there as a new function.

Reply:
Sorry, I meant `pkcs2tee_proc_params_rsa_pss()` could do all the verification below, it only needs the modules bit size as extra argument.

Reply:
Now there is to functions. Please check it out.

Can merge the two if you feel it is better that way.

I kinda feel that parsing and translating the structure should have own method and no duplication of that code elsewhere.